### PR TITLE
Add composer validate check to CI

### LINKED ISSUE #23930: CI must warn if composer.lock does not satisfy composer.json
#### Describe the bug

see title

#### Expected behavior

CI must detect the issue

#### Steps to Reproduce

see https://github.com/PrestaShop/PrestaShop/pull/23896/commits/57386655354d935f013b2bcf55d708604ee41a1f and related CI results

#### Additional information

* PrestaShop version: develop

### PR DESCRIPTION
**Pull request by mvorisek**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add composer validate check to CI
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23930
| How to test?      | GH CI must pass
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23931)

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "description": "PrestaShop offers a free, fully scalable, Open Source e-commerce solution.",
   "type": "project",
   "require": {
-    "php": ">=7.1.3",
+    "php": "^7.1.3",
```
Comment (ks129) [inline composer.json:6]: I don't think this change is right. This should support PHP versions higher than `7.1.3`.

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -25,6 +25,9 @@ jobs:
               restore-keys: |
                   ${{ runner.os }}-composer-
 
+      -   name: Validate composer.{json,lock} files
+          run: composer validate --strict
+
```
Comment (ks129) [inline .github/workflows/php.yml]: This shouldn't be part of PHP CS fixer, but a separate job instead.

Comment (mvorisek) [reply .github/workflows/php.yml]: I can do that, but I do not think extra job is good for this small check.

PS Maintaners, advise, I can do that if you want.

File: composer.json (hunk 1 above)
Comment (mvorisek) [reply composer.json:6]: This is exactly what it does, it allows PHP >= `7.1.3` < `8.0`

Comment (ks129) [reply composer.json:6]: Thanks for the explanation, caret symbol's meaning in PHP is different than in other languages.

**Comment by mvorisek** [comment]
@eternoendless this PR prevents issues like https://github.com/PrestaShop/PrestaShop/pull/23950 , please merge if ok, I rebased it onto develop now

**Review by eternoendless — APPROVED** [review]
Good idea!

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @mvorisek  :wink:

**Pull request merged**